### PR TITLE
Fix #196: Integrate cppnetlib as a git submodule

### LINKED ISSUE #196: git_external for ccp-netlib
Hi Guys,

I was upgrading the nix builds for ZeroEQ and I have seen that you sneak-in one git_external dependency.

git_external(${COMMON_SOURCE_DIR}/cppnetlib
https://github.com/BlueBrain/cpp-netlib 0.11-bbp)

Please, could you provide a way to disable that and build **without** git_external,  by using a cppnetlib installation path manually provided through CMAKE_PREFIX_PATH like it works for subprojects.

Like I already said, no decent  package manager would accept a software to download random staff during configure phase. This is a blocker for me to Upgrade ZeroEQ, and the upgrade is necessary for other package due to API changes.

Cheers,
Adrien

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by rdumusc** [comment]
+1

**Comment by hernando** [comment]
Again late. GitExternal.cmake gets automatically updated by CMake/common, you need to update it also there.

**Comment by tribal-tec** [comment]
GitExternal.cmake is updated?! And you're not too late. Waiting on travis...

**Comment by hernando** [comment]
I don't know from where and when that's done, but as far as I know, yes, it happens.

**Comment by hernando** [comment]
Confirmed: UpdateGitExternal.cmake, which is included from Common.cmake

**Comment by tribal-tec** [comment]
Approve anyone?

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,13 +32,8 @@ set(ZEROEQ_MAINTAINER "Human Brain Project <HBPVis@googlegroups.com>")
...
 set(ZEROEQ_DEPENDENT_LIBRARIES Servus)
 
-# enforce cloning our own version of cppnetlib and skip finding a potentially
-# installed system package
-git_external(${COMMON_SOURCE_DIR}/cppnetlib
-  https://github.com/BlueBrain/cpp-netlib 0.11-bbp)
-add_subproject(cppnetlib)
-
 common_find_package(Boost COMPONENTS unit_test_framework)
+add_subproject(cppnetlib ${PROJECT_SOURCE_DIR}/cppnetlib) # from .gitmodules
```
Comment (hernando) [inline CMakeLists.txt]: I'd still keep it separate at the top as before, this line looks out of place to me.

**Pull request merged**